app/main.py

```diff
@@ -164,6 +164,40 @@ def sim_dataset(id):
         return jsonify(json)
 
 
+@app.route("/search/<query>")
+def kb_search(query):
+    try:
+        response = requests.get(f'https://scicrunch.org/api/1/elastic/SPARC_Datasets_pr/_search?q={query}&api_key={Config.KNOWLEDGEBASE_KEY}')
+        return process_kb_results_recursive(response.json())
+    except requests.exceptions.HTTPError as err:
+        logging.error(err)
+        return json.dumps({'error': err})
+
+@app.route("/search/")
```

Comment:
It is possible to have the query argument here as an optional, that would merge these two methods into one. 
[https://flask.palletsprojects.com/en/master/api/#url-route-registrations](url)
[https://stackoverflow.com/questions/53445145/flask-optional-path-args](url)

Reply:
Oo thanks Alan! I didn't think to try checking if that was possible.